crates/bevy_ecs/src/system/system_name.rs

```diff
@@ -131,4 +131,14 @@ mod tests {
         let name = world.run_system(id).unwrap();
         assert!(name.ends_with("testing"));
     }
+
+    #[test]
+    fn test_closure_system_name_regular_param() {
+        let mut world = World::default();
+        let id = world.register_system(
+            IntoSystem::into_system(|name: SystemName| name.name().to_owned()).with_name("testing"),
+        );
+        let name = world.run_system(id).unwrap();
```

Comment:
This can just be world.run_system_once, which saves us doing the system registration dance.